### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by nilab-khwaja**
**### Splash screen**

- A simple page with the name of  app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
**- Home page (categories page)**

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
**- Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
**- "Add a new category" page**

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
**- "Add a new transaction" page**

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.
- Technical requirements
- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:
Capture

### CODE REVIEW

File: app/views/categorys/index.html.erb
Comment (Efiamotu-1) [inline app/views/categorys/index.html.erb:1]: - [ ] From the images below you can notice that your design for the category page is inconsistent with the provided project design, the Add new category button should be permanently fixed to the bottom end of the screen as shown in the image below, please kindly adjust your project to match this design :+1: 
- [ ] Kindly make sure to add this change as well in the transaction page :+1: :sparkles: 
- project design
cat
- your design
ADD

Comment (Efiamotu-1) [inline app/views/categorys/index.html.erb:1]: - [ ] This application is a mobile app however when i set the screen size to view as a mobile app the design doesn't match correctly. please kindly adjust the design appropriately :+1: 
mb

Comment (Efiamotu-1) [inline app/views/categorys/index.html.erb:1]: - [ ] From the project design below, each category should be displayed one above the other, however from your image design after adding three categories they are all showing side by side, please kindly fix this :+1: 
- project design
1ab
- your design
3ab
- [ ] Also while trying to add a new category it requires an icon but when i put in a link to an icon from the internet it doesn't come up.

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @nilab-khwaja, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nilab-khwaja** [comment]
Dear Reviewer , Kindly use this links for adding icon to the category .
https://robohash.org/perspiciatisremrepellat.png
 https://robohash.org/odiovelrerum.png
https://robohash.org/estconsequunturcorporis.png
 I already test and it looks like this :
Capture

File: app/views/transactions/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<%=render "shared/navbar", right_el: "menu", left_el: "none", title: "TRANSACTIONS"%>
+<div class="bg-secondary text-light"><span>Total Amount: $100, Total Items: 9</span></div>
+    <ul class="category-list d-flex flex-column align-items-center m-auto  p-3">
+        <%for a in 0..30 do%>
+            <%= render "transactions/shared/transaction"%>
+        <%end%>
+    </ul>
```
Comment (thecodechaser) [inline app/views/transactions/index.html.erb]: - [ ] Great job fixing the buttons in the app as requested by the previous reviewer but there's a issue in the transaction page, If I add some transactions and they exceed the height of the page I'm not able to see the total amount it's underflowing the `add new transaction` button. Please make sure to fix this issue (You can reference the categories page that is working fine in the same situation.

image

Comment (thecodechaser) [inline app/views/transactions/index.html.erb]: - [ ] [Optional] This change was missed by the previous reviewer so I'm marking it as optional but I would highly recommend to implementing it for a better user experience. 
I would suggest in the transaction page to show the total amount at the top of the page as requested in the project requirements, Currently it's showing at the bottom of the page.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @nilab-khwaja👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Nice UI design. ✔️ 
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @nilab-khwaja,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Descriptive PR⭐
- [ ]  No Linters error 🤩
- [ ] Good job working on the controllers💪🏻
- [ ] Amazing job working on the models 🔥🔥
- [ ] Great for making the requested changes ⭐
- [ ] Your design is great 🔥
- [ ] Project deployed well ⭐


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**